### PR TITLE
Refactor record marshaling and processing in GDNS publisher

### PR DESCRIPTION
**Pull request by ocsw**
**What this PR does / why we need it**:
We need to separate the marshaling of resource records and the processing of each record so we can change the marshaling in subclasses.

**Which issue(s) this PR fixes**

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Refactored GDNSPublisher for easy subclassing/overriding.
```

### CODE REVIEW

**Comment by ocsw** [comment]
@spotify/alf ok?  This is a prereq for https://ghe.spotify.net/gordon/gordon-internal/pull/9.

File: src/gordon_gcp/plugins/service/gdns_publisher.py [hunk 1]
```diff
@@ -298,6 +299,38 @@ def _format_resource_record_changes(self, action, resource_record):
 
         return False
 
+    async def _process_resource_record(self, resource_record, action, logger):
+        """Publish changes for one recordd, making sure they are completed.
```
Comment (descent3me) [inline src/gordon_gcp/plugins/service/gdns_publisher.py]: typo

**Comment by ocsw** [comment]
@descent3me You wanted me to change the name of the factored out function.  How's this?

File: src/gordon_gcp/plugins/service/gdns_publisher.py [hunk 2]
```diff
@@ -298,6 +299,39 @@ def _format_resource_record_changes(self, action, resource_record):
 
         return False
 
+    async def _dispatch_changes(
+            self, resource_record, action, logger):
```
Comment (descent3me) [inline src/gordon_gcp/plugins/service/gdns_publisher.py]: I know its defined in the spec, but I'd prefer is we could do the line breaking at least to a de-facto pattern. I think we tend to start listing the args on the same line as the method declaration.

Comment (matt0bi) [reply src/gordon_gcp/plugins/service/gdns_publisher.py]: ... mumble [yapf](https://github.com/google/yapf) mumble mumble ... :)

Comment (ocsw) [reply src/gordon_gcp/plugins/service/gdns_publisher.py]: Ha, it actually fits on one line now.

**Comment by descent3me** [comment]
+1

**Pull request merged**